src/experimental/test/property/arbitraries/canister_arb.ts

```diff
@@ -145,26 +144,28 @@ function generateSourceCode<
     ParamAgentResponseValue
 >(
     globalDeclarations: string[],
-    canisterMethods: (
-        | UpdateMethod<ParamAgentArgumentValue, ParamAgentResponseValue>
-        | QueryMethod
-    )[],
-    api: Api,
+    canisterMethods: CanisterMethod<
```

Comment:
Let's discuss this

Reply:
these types were wrong and caused me much confusion. Its really a list of all of the canister methods and not just query and updates as the types would have you believe. this error was discovered upon removing the global declarations from inspect message and preUpgrade